src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/NetworkFormula.kt

```diff
@@ -34,7 +35,22 @@ class NetworkFormula(
             stack.findVpc("Vpc"),
             stack.findSubnet("TheOnlySubnet")
         )
-        logger.info("Network provisioned: $network")
-        return network
+        logger.info("Network provisioned")
+        logger.debug("Network provisioned: $network")
+        return ProvisionedNetwork(network, stack)
     }
+
+    internal fun reuseOrProvision(existing: Network?) : ProvisionedNetwork {
```

Comment:
I'll be honest that I don't like the idea that this method exists. Good that it's `internal`, so that we can refactor it later.

I don't want to go deeper why existance of this method is a problem, because there is a design level flaw one step higher. I think we should at some point get rid of the hacky "network override" mechanism done inside `InfrastructureFormula`. It relies on `instanceof`, which makes it tricky to use custom implementations of `JiraFormula`.

Hopefully API redesign as part of https://ecosystem.atlassian.net/browse/JPERF-273 will cover this problem.

Reply:
True, it's smelly. The network feature added this complication due to compatibility. It should be simplified in the next major release. It's now easier to track this smell, because it's extracted and named.